pkgs/development/compilers/gcc/ng/common/gcc/default.nix

```diff
@@ -0,0 +1,246 @@
+    #
+    # This is all 3 commits squashed.
+    (getVersionFile "gcc/find-prefixed-progs.patch")
+    (getVersionFile "gcc/custom-threading-model.patch")
+
+    # TODO: fix up and send to upstream
+    (getVersionFile "gcc/fix-collect2-paths.diff")
+  ];
+
+  enableParallelBuilding = true;
+
+  hardeningDisable = [
+    "format" # Some macro-indirect formatting in e.g. libcpp
+  ];
+
+  strictDeps = true;
+
+  depsBuildBuild = [ buildPackages.stdenv.cc ];
+  nativeBuildInputs = [
+    texinfo
+    which
+    gettext
+  ] ++ lib.optional (perl != null) perl;
+
+  buildInputs =
+    [
+      gmp
+      libmpc
+      mpfr
+      libelf
+    ]
+    ++ lib.optional (isl != null) isl
+    ++ lib.optional (zlib != null) zlib;
+
+  postUnpack = ''
+    mkdir -p ./build
+    buildRoot=$(readlink -e "./build")
+  '';
+
+  postPatch =
+    ''
+      configureScripts=$(find . -name configure)
+      for configureScript in $configureScripts; do
+        patchShebangs $configureScript
+      done
+
+      patchShebangs libbacktrace/install-debuginfo-for-buildid.sh
+      patchShebangs runtest
+    ''
+    # This should kill all the stdinc frameworks that gcc and friends like to
+    # insert into default search paths.
+    + lib.optionalString hostPlatform.isDarwin ''
+      substituteInPlace gcc/config/darwin-c.c \
+        --replace 'if (stdinc)' 'if (0)'
+    '';
+
+  preConfigure =
+    # Don't built target libraries, because we want to build separately
+    ''
+      substituteInPlace configure \
+        --replace 'noconfigdirs=""' 'noconfigdirs="$noconfigdirs $target_libraries"'
+    ''
+    # HACK: if host and target config are the same, but the platforms are
+    # actually different we need to convince the configure script that it
+    # is in fact building a cross compiler although it doesn't believe it.
+    +
+      lib.optionalString (targetPlatform.config == hostPlatform.config && targetPlatform != hostPlatform)
+        ''
+          substituteInPlace configure --replace is_cross_compiler=no is_cross_compiler=yes
+        ''
+    # Cannot configure from src dir
+    + ''
+      cd "$buildRoot"
+
+      mkdir -p "$buildRoot/libbacktrace/.libs"
+      cp ${buildGccPackages.libbacktrace}/lib/libbacktrace.a "$buildRoot/libbacktrace/.libs/libbacktrace.a"
+      cp -r ${buildGccPackages.libbacktrace}/lib/*.la "$buildRoot/libbacktrace"
+      cp -r ${buildGccPackages.libbacktrace.dev}/include/*.h "$buildRoot/libbacktrace"
+
+      mkdir -p "$buildRoot/libiberty/pic"
+      cp ${buildGccPackages.libiberty}/lib/libiberty.a "$buildRoot/libiberty"
+      cp ${buildGccPackages.libiberty}/lib/libiberty_pic.a "$buildRoot/libiberty/pic/libiberty.a"
+      touch "$buildRoot/libiberty/stamp-noasandir"
+      touch "$buildRoot/libiberty/stamp-h"
+      touch "$buildRoot/libiberty/stamp-picdir"
+
+      mkdir -p "$buildRoot/build-${stdenv.hostPlatform.config}"
+      cp -r "$buildRoot/libiberty" "$buildRoot/build-${stdenv.hostPlatform.config}/libiberty"
```

Comment:
Did we have this before? What exactly is going on? Seems like this should be not `buildGccPackages` but the same package set as GCC itself? At least if GCC is going to link these?

Is this the top-level configure you are using? can we use a GCC subdir instead to avoid some of this?

Reply:
OK that sounds good. I see comparing to the earlier versions of this that this is not a different approach but just added stuff to the same approach -- I was worried you might be doing this instead of just CD'ing to a subproject, but I see that is not the case.